src/components/map/hooks/use-map-symbol-styles.ts

```diff
@@ -0,0 +1,185 @@
+import {useThemeContext} from '@atb/theme';
+
+import {nsrSymbolLayers} from '../components/national-stop-registry-features/nsr-layers';
+import {NsrProps} from '../components/national-stop-registry-features/NationalStopRegistryFeatures';
+import {
+  ExpressionField,
+  Expression,
+  SymbolLayerStyleProps,
+} from '@rnmapbox/maps/src/utils/MapboxStyles';
+import {PinType} from '../mapbox-styles/pin-types';
+
+// Returns Mapbox Style Expressions to determine map symbol styles.
+export const useMapSymbolStyles = (
+  selectedFeaturePropertyId: NsrProps['selectedFeaturePropertyId'],
+  pinType: PinType,
+  textSizeFactor: number = 1.0,
+) => {
+  const {themeName} = useThemeContext();
+  const isDarkMode = themeName === 'dark';
+
+  const featureId: Expression = ['get', 'id'];
+  const selectedFeatureId = selectedFeaturePropertyId || ''; // because mapbox style expressions don't like undefined
+  const isSelected: Expression = ['==', featureId, selectedFeatureId];
+
+  const countPropName = 'point_count'; // switch to 'count' when using vector source for shared mobility items
+  const count: Expression = ['get', countPropName];
+  const numVehiclesAvailable: Expression = ['get', 'count']; // switch to 'num_vehicles_available' when using vector source for shared mobility items
+
+  const isCluster: Expression = [
+    'all',
+    ['has', countPropName],
+    ['!=', count, 1],
+  ];
+
+  const mapItemIconNonClusterState: Expression = [
+    'case',
+    isSelected,
+    'selected',
+    ['case', ['==', selectedFeatureId, ''], 'default', 'minimized'],
+  ];
+  const mapItemIconState: Expression = [
+    'case',
+    isCluster,
+    'cluster',
+    mapItemIconNonClusterState,
+  ];
+
+  const iconSize: Expression = [
+    'case',
+    pinType === 'station' ? true : isCluster,
+    0.855,
+    1,
+  ];
+
+  const stopPlacesExpression: (Expression | ExpressionField)[] = nsrSymbolLayers
+    .filter(
+      (nsrSymbolLayer) =>
+        nsrSymbolLayer.filter !== undefined &&
+        nsrSymbolLayer.iconCode !== undefined,
+    )
+    .flatMap((nsrSymbolLayer) => [
+      nsrSymbolLayer.filter,
+      nsrSymbolLayer.iconCode as ExpressionField,
+    ]);
+
+  const vehicle_type_form_factor: Expression = [
+    'get',
+    'vehicle_type_form_factor',
+  ];
+  const iconCode: Expression = [
+    'case',
+    ...stopPlacesExpression,
+    ['==', vehicle_type_form_factor, 'SCOOTER'],
+    'scooter',
+    ['==', vehicle_type_form_factor, 'BICYCLE'],
+    'citybike',
+    ['==', vehicle_type_form_factor, 'CAR'],
+    'sharedcar',
+    'non-existing-icon',
+  ];
+
+  const systemId: Expression = ['get', 'system_id'];
+  const transportOperator: Expression = [
+    'case',
+    ['!', ['has', 'system_id']],
+    'generic',
+    [
+      'case',
+      ['==', ['slice', systemId, 0, 4], 'ryde'],
```

Comment:
can those operators be taken for a dynamic list of operators, maybe later we want to turn some of them off.